m4/liboauthcpp.m4

```diff
@@ -0,0 +1,23 @@
+AC_DEFUN([LIBOAUTHCPP_INIT],[
+  AC_SUBST(HAS_LIBOAUTHCPP) 
+
+  AC_LANG_PUSH([C++])
+
+  CPPFLAGS="-I/usr/liboauthcpp/liboauthcpp.h ${CPPFLAGS}"
+  LIBS="-loauthcpp -L/usr/lib64 ${LIBS}"
+
+  AC_CHECK_HEADERS(
+    [liboauthcpp/liboauthcpp.h], 
+    [hootFoundLiboauthcppHeaders=yes; break;], 
+    [AC_MSG_FAILURE("Unable to find liboauthcpp header"); break;])
+
+  AC_LINK_IFELSE(
+    [AC_LANG_PROGRAM(
+      [[#include <liboauthcpp/liboauthcpp.h>]],
+      [[OAuth::Client::initialize();]])],
+    [],
+    [hootFoundLiboauthcppHeaders=no; AC_MSG_FAILURE("Unable to link to liboauthcpp"); break;])
+
+  AC_LANG_POP
+])
+
```

Comment:
Mostly a copy of other `.m4` files to add the `-loauthcpp` switch to the linker, verify the `<liboauthcpp/liboauthcpp.h>` file is found, and create a sample program that includes that header and calls a static `OAuth::Client::initialize()` function.